parser/prism/Translator.cc

```diff
@@ -939,9 +939,23 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
             auto name = translate(moduleNode->constant_path);
             auto declLoc = translateLoc(moduleNode->module_keyword_loc).join(name->loc);
 
-            auto body = translate(moduleNode->body);
+            auto moduleTranslator = enterModule();
+            auto bodyNode = moduleTranslator.translate(moduleNode->body);
 
-            return make_unique<parser::Module>(location, declLoc, move(name), move(body));
+            if (!hasExpr(name)) {
+                return make_unique<parser::Module>(location, declLoc, move(name), move(bodyNode));
+            }
+
+            auto bodyStore = bodyToRHSStore(bodyNode);
+            if (!bodyStore.has_value()) {
+                return make_unique<parser::Module>(location, declLoc, move(name), move(bodyNode));
+            }
+
+            auto nameExpr = name->takeDesugaredExpr();
+            ast::ClassDef::ANCESTORS_store ancestors;
+            auto expr = MK::Module(location, declLoc, std::move(nameExpr), std::move(ancestors), std::move(bodyStore.value()));
+
+            return make_node_with_expr<parser::Module>(std::move(expr), location, declLoc, move(name), move(bodyNode));
```

Comment:
Really wishing C++ had an equivalent to [Rust's `let`-`else`](https://doc.rust-lang.org/beta/rust-by-example/flow_control/let_else.html) or [Swift's `guard`](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/controlflow/#Early-Exit) statements ```suggestion

Is nesting this into the `if (bodyStore.has_value())` any better? 🤔 

```c++
            if (bodyStore.has_value()) {
                return make_unique<parser::Module>(location, declLoc, move(name), move(bodyNode));
                auto nameExpr = name->takeDesugaredExpr();
                ast::ClassDef::ANCESTORS_store ancestors;
                auto expr = MK::Module(location, declLoc, move(nameExpr), move(ancestors), move(*bodyStore));

                return make_node_with_expr<parser::Module>(std::move(expr), location, declLoc, move(name), move(bodyNode));
            } else {
                 return make_unique<parser::Module>(location, declLoc, move(name), move(bodyNode));
            }
```

You can't bind the unwrapped optional, so you still need `*bodyStore` :(